src/main/kotlin/com/openlattice/BackgroundExternalDatabaseSyncingService.kt

```diff
@@ -142,7 +142,26 @@ class BackgroundExternalDatabaseSyncingService(
             }
 
         }
-        
+
+        //check if tables have been deleted in the database
+        val missingTableIds = organizationExternalDatabaseTables.keys - currentTableIds
+        if (missingTableIds.isNotEmpty()) {
+            edms.deleteOrganizationExternalDatabaseTableObjects(missingTableIds)
+            totalSynced += missingTableIds.size
+        }
+
+        //check if columns have been deleted in the database
+        val missingColumnIds = organizationExternalDatabaseColumns.keys - currentColumnIds
+        val missingColumnsByTable = mutableMapOf<UUID, MutableSet<UUID>>()
+        if (missingColumnIds.isNotEmpty()) {
+            missingColumnIds.forEach {
+                val tableId = organizationExternalDatabaseColumns.getValue(it).tableId
```

Comment:
Actually there's an easy perf improvement we can make here that's super simple -- if we switch this to a bulk call outside of the forEach, then this won't have to make a separate network call for each missing column id. Something like
```
if (missingColumnIds.isNotEmpty()) {
  organizationExternalDatabaseColumns.getAll(missingColumnIds).forEach { (tableId, column) ->
    val tableId = column.tableId
    missingColumnsByTable.getOrPut(tableId) { mutableSetOf() }.add(it)
  }
  ...
```

Reply:
...and if you wanted to be `~super~` kotliny + shorten this, you could do something like this...

```
val missingColumnsByTable = organizationExternalDatabaseColumns.getAll(missingColumnIds).entries
                .groupBy { it.value.tableId }
                .mapValues { it.value.map { entry -> entry.key } }
```